### PR TITLE
Update tests starting with letters (R-Z) to use assertj 

### PR DESCRIPTION
**Pull request by pavanbaloju**
# pull request for (#2147 )

In this PR, the following tests have been updated to use assertj
1. RailFenceCipherTest.java
2. RaindropConverterTest.java
3. RationalTest.java
4. RectangleCounterTest.java
5. RestApiTest.java
6. ReverseStringTest.java
7. RobotTest.java
8. RomanNumeralsTest.java
9. RunLengthEncodingTest.java
10. MatrixTest.java
11. ScrabbleScoreTest.java
12. SeriesTest.java
13. SieveTest.java
14. SimpleCipherStepOneTest.java
15. SimpleCipherStepTwoSubsitutionTest.java
16. SpaceAgeTest.java
17. SpiralMatrixBuilderTest.java
18. RelationshipComputerTest.java
19. SumOfMultiplesTest.java
20. TournamentTest.java
21. TwelveDaysTest.java
22. TwoBucketTest.java
23. VariableLengthQuantityTest.java
24. WordCountTest.java
25. WordSearcherTest.java
26. WordProblemSolverTest.java
27. YachtTest.java
28. ZebraPuzzleTest.java

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/main/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by pavanbaloju** [comment]
Hi @sanderploegsma 
One last PR from my side for the remaining classes. Please review!

File: exercises/practice/simple-cipher/src/test/java/SimpleCipherStepOneTest.java [hunk 1]
```diff
@@ -26,27 +25,27 @@ public void setup() {
...
-        assertEquals(plainText, cipherWithDefaultKey.decode(cipherWithDefaultKey.encode(plainText)));
+        assertThat(cipherWithDefaultKey.decode(cipherWithDefaultKey.encode(plainText))).isEqualTo(plainText);
     }
 
     @Ignore("Remove to run test")
     @Test
     public void keyIsLowercaseLetters() {
         String key = cipherWithDefaultKey.getKey();
-        assertTrue(Pattern.matches("^[a-z]+$", key));
+        assertThat(Pattern.matches("^[a-z]+$", key)).isTrue();
```
[the remark this answers is not in the document]
Comment (pavanbaloju) [reply exercises/practice/simple-cipher/src/test/java/SimpleCipherStepOneTest.java:49]: yeah we can improvise it

[the remark this answers is not in the document]
Comment (pavanbaloju) [reply exercises/practice/simple-cipher/src/test/java/SimpleCipherStepOneTest.java]: Done

[the remark this answers is not in the document]
Comment (sanderploegsma) [reply exercises/practice/simple-cipher/src/test/java/SimpleCipherStepOneTest.java]: Great, thanks!

**Comment by sanderploegsma** [comment]
Thanks again!

**Comment by pavanbaloju** [comment]
Thanks and welcome!

**Pull request merged**